### PR TITLE
Honor `--config` flag as save location for `config save`

### LINKED ISSUE #986: "overwrite file" warning on `config save` ignores `--config` option
**Describe the bug**

it looks like the "an existing configuration file has been found. Overwrite file" warning ignores the `--config` cli option.

In the execution pasted below, I have a fly.toml, but I do not have a fly-db.toml. I want to create a fly-db.toml with `config save`, but it warns me that I'd be overwriting `fly.toml`, but after I confirm, it does not overwrite fly.toml.

* Operating system: macos 12.4
* `fly version`: `fly v0.0.329 darwin/amd64 Commit: 89104d7 BuildDate: 2022-05-25T16:50:30Z`

** Command output: **

```text
> fly config save --app=amonks-request-testing-db --config=fly-db.toml
***An existing configuration file has been found.
? Overwrite file '/Users/ajm/request/fly.toml' Yes
Wrote config file fly-db.toml


> git status
On branch master
Your branch is ahead of 'origin/master' by 1 commit.
  (use "git push" to publish your local commits)

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        fly-db.toml

nothing added to commit but untracked files present (use "git add" to track)
```

### PR DESCRIPTION
**Pull request by ndarilek**
Fixes #986

### CODE REVIEW

**Comment by ndarilek** [comment]
Good question. I'm inclined to say no. If I remember this PR correctly, 
it's saving configuration from an existing app. If you're specifying 
your own path for the save, it may be because you don't want to 
overwrite whatever's at the existing path. Then again, maybe not, but in 
that case I'd guess the user wants to debug what failed vs. suddenly 
discovering their original config file got rewritten. Happy to consider 
other perspectives.

File: cmd/config.go [hunk 1]
```diff
@@ -55,10 +55,7 @@ func runShowConfig(cmdCtx *cmdctx.CmdContext) error {
 func runSaveConfig(cmdCtx *cmdctx.CmdContext) error {
 	ctx := cmdCtx.Command.Context()
 
-	configfilename, err := flyctl.ResolveConfigFileFromPath(cmdCtx.WorkingDir)
-	if err != nil {
-		return err
-	}
+	configfilename := cmdCtx.ConfigFile
```
[the remark this answers is not in the document]
Comment (ndarilek) [reply cmd/config.go:58]: Ack, sent an email reply to this but am not sure if it got correctly associated with this review.

Happy to revisit it if I made the wrong call here, but I'd like to merge or close it soon. Thanks!

**Comment by billyb2** [comment]
@ndarilek i personally think not overwriting existing configs is safest, but take my opinion with a grain of salt.

**Comment by ndarilek** [comment]
Circling back to this:

I'd like to resolve it one way or the other--either honor the `-c` flag as a save location, or remove it entirely so there's no ambiguity. But does either make sense? Is `-c` used to target the remote service for retrieving the configuration to save? Maybe we need a separate output flag to make it clear what's what?

**Comment by dangra** [comment]
hello! this command was moved to https://github.com/superfly/flyctl/blob/4a91f721f1d66c8eeaa4f7259ca16c66595f9d6e/internal/command/config/save.go

I think we use current-working-dir as path unless `--config` is passed 

https://github.com/superfly/flyctl/blob/4a91f721f1d66c8eeaa4f7259ca16c66595f9d6e/internal/command/config/save.go#L36-L37 

on the lines of:
```go
path := state.WorkingDirectory(ctx)
if flag.IsSpecified("config") {
  path = flag.GetString("config")
}
configfilename, err := appconfig.ResolveConfigFileFromPath(path)
...
```

As a semi off topic bugfix, call `ResolveConfigFilepath()` on the [joined path](https://github.com/superfly/flyctl/blob/4a91f721f1d66c8eeaa4f7259ca16c66595f9d6e/internal/appconfig/file.go#L26) to prevent overwriting an existent file.

**Comment by ndarilek** [comment]
Following up on yesterday's confusion:

It does appear that some context data, including app name, is optionally read from fly.toml in the current directory. In order to not muddy the waters, I added a separate `-o` flag. With this, `fly config save -o new.toml` saves the config from the specified app into a different config file.

**Comment by dangra** [comment]
> I added a separate `-o` flag. With this, `fly config save -o new.toml` saves the config from the specified app into a different config file.

hmm, I am +1 to use the same flag that is already there, otherwise command's usage message becomes messy:

```
l$ fly config save --help
Save an application's configuration locally. The configuration data is
retrieved from the Fly service and saved in TOML format.

Usage:
  flyctl config save [flags]

Flags:
  -a, --app string      Application name
  -c, --config string   Path to application configuration file
  -h, --help            help for save
  -o, --output string   Path to output application configuration file

Global Flags:
  -t, --access-token string   Fly API Access Token
  -j, --json                  json output
      --verbose               verbose output
```

there is no harm in reusing the the value of `--config` as long as it checks the file exists and ask to override.
In fact, if you run `fly config save` in a directory with an existent fly.toml, it will do the same as calling it with `fly config save --config fly.toml` 

If you also address this past comment, there is no chance an existing fly.toml can be overridden without asking the user: 

> As a semi off topic bugfix, call ResolveConfigFilepath() on the [joined path](https://github.com/superfly/flyctl/blob/4a91f721f1d66c8eeaa4f7259ca16c66595f9d6e/internal/appconfig/file.go#L26) to prevent overwriting an existent file.

**Comment by ndarilek** [comment]
Ah, sorry, somehow I missed the suggestion about `ResolveConfigPath`. Going to finish this one off in a bit.

**Comment by ndarilek** [comment]
I think this should do it. Apologies for misunderstanding/missing the review comments.

Tested scenarios:
* `fly config save -a ...` writes to fly.toml by default.
* `fly config save -a ... --config new.toml` writes config to new.toml.
* `fly config save -a ... --config existing_fly.toml` prompts before overwriting.

**Review by dangra — APPROVED** [review]
thanks @ndarilek !

**Pull request merged**